File: MealPlanService/MealPlanService.Application/DTOs/Creating/MealPlanCreateDTO.cs
using MealPlanService.Domain.Enums;
using Microsoft.AspNetCore.Http;


namespace MealPlanService.Application.DTOs
{
    public class MealPlanCreateDTO
    {
        public int OwnerId { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public MealPlanType Type { get; set; }
        public List<MealPlanDayDTO> Days { get; set; } = new List<MealPlanDayDTO>();
    }
}

File: NutritionBalance.Client/src/app/models/mealPlan/DTOs/meal-plan-create.dto.ts
import { MealPlanDayDTO } from './meal-plan-day.dto';
import { MealPlanType } from '../enums/meal-plan-type.enum';

export class MealPlanCreateDTO {
  ownerId!: number;
  name!: string;
  description!: string;
  type!: MealPlanType;
  days: MealPlanDayDTO[] = [];
}
